### PR TITLE
feat: migrate to tauri 2.0

### PR DESCRIPTION
**Pull request by brayo-pip**
> [!IMPORTANT]
> Migrate project to Tauri 2.0, updating dependencies, refactoring code, and adding new capabilities and permissions.
> 
>   - **Migration to Tauri 2.0**:
>     - Update `package.json` dependencies to Tauri 2.0 and related plugins.
>     - Update `Cargo.lock` with new versions of dependencies.
>     - Modify `tauri.conf.json` to use Tauri 2.0 schema.
>   - **Code Refactoring**:
>     - Move main logic from `main.rs` to `lib.rs` and update `main.rs` to call `aw_tauri_lib::run()`.
>     - Refactor tray menu setup in `manager.rs` to use Tauri 2.0 APIs.
>   - **New Features**:
>     - Add new capabilities and permissions in `src-tauri/capabilities` and `src-tauri/gen/schemas`.
>     - Implement new `greet` command in `lib.rs` and update `App.vue` to use it.
>   - **Build and Configuration**:
>     - Update `vite.config.ts` for Tauri 2.0 compatibility.
>     - Modify `tsconfig.node.json` for module resolution changes.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ActivityWatch%2Faw-tauri&utm_source=github&utm_medium=referral)<sup> for d92dfad1c2181f8012069456215cf81164f9a30c. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by brayo-pip** [comment]
Most of these are failing because of a new library that needs a custom_build command for.

File: src-tauri/Cargo.toml [hunk 1]
```diff
@@ -1,21 +1,29 @@
 [package]
 name = "aw-tauri"
-version = "0.0.0"
+version = "0.1.0"
 description = "ActivityWatch UI built with Tauri"
-authors = ["Erik Bjäreholt <erik@bjareho.lt>"]
+authors = ["Erik Bjäreholt <erik@bjareho.lt> Brian Vuku <brayo@brayo.dev>"]
```
Comment (ErikBjare) [inline src-tauri/Cargo.toml:5]:
```suggestion
authors = ["Erik Bjäreholt <erik@bjareho.lt>", "Brian Vuku <brayo@brayo.dev>"]
```

File: src-tauri/gen/schemas/capabilities.json
Comment (ErikBjare) [inline src-tauri/gen/schemas/capabilities.json:1]: Do these schemas need to be committed?

File: src-tauri/gen/schemas/desktop-schema.json
Comment (ErikBjare) [inline src-tauri/gen/schemas/desktop-schema.json:1]: Seems to be a duplicate of linux-schema.json?

**Review by ErikBjare — COMMENTED** [review]
Hard to review, but looks good! 
What's up with all the schemas?

**Comment by ErikBjare** [comment]
Ubuntu build fails with:

```
The following warnings were emitted during compilation:

warning: glib-sys@0.18.1: 

error: failed to run custom build command for `glib-sys v0.18.1`

Caused by:
  process didn't exit successfully: `/home/runner/work/aw-tauri/aw-tauri/src-tauri/target/release/build/glib-sys-2af436f5e2da36ff/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=GLIB_2.0_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:warning=
  pkg-config exited with status code 1
  > PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 pkg-config --libs --cflags glib-2.0 glib-2.0 >= 2.70

  The system library `glib-2.0` required by crate `glib-sys` was not found.
  The file `glib-2.0.pc` needs to be installed and the PKG_CONFIG_PATH environment variable must contain its parent directory.
  The PKG_CONFIG_PATH environment variable is not set.

  HINT: if you have installed the library, try setting PKG_CONFIG_PATH to the directory containing `glib-2.0.pc`.

warning: build failed, waiting for other jobs to finish...
failed to build app: failed to build app
    Error failed to build app: failed to build app
Error: Command failed with exit code 1: npm run tauri build
```

I guess it needs some glib apt package?

File: src-tauri/gen/schemas/capabilities.json
Comment (brayo-pip) [reply src-tauri/gen/schemas/capabilities.json:1]: They seem to be autogenerated. No need to track. I deleted them and `make dev` runs. `make build` regenerates them.

Comment (brayo-pip) [reply src-tauri/gen/schemas/capabilities.json:1]: Tauri website says they are for autocompletion in IDE. [link](https://tauri.app/security/capabilities/#schema-files). Let's keep them for now.

**Comment by brayo-pip** [comment]
Will try fixing the tests tonight

**Comment by brayo-pip** [comment]
@ErikBjare fixed the tests :). Got caught up last week, but I am back :+1:

**Comment by ErikBjare** [comment]
Nice! Let's go 🚀

**Pull request merged**